xposed-bridge/src/main/java/de/robv/android/xposed/XSharedPreferences.java

```diff
@@ -50,36 +51,19 @@
     private boolean mLoaded = false;
     private long mLastModified;
     private long mFileSize;
-    private boolean mWatcherEnabled;
-
-    private static synchronized WatchService getWatcher() {
-        if (sWatcher == null) {
-            try {
-                sWatcher = new File(XposedInit.prefsBasePath).toPath().getFileSystem().newWatchService();
-                if (BuildConfig.DEBUG) Log.d(TAG, "Created WatchService instance");
-            } catch (IOException e) {
-                Log.e(TAG, "Failed to create WatchService", e);
-            }
-        }
-
-        if (sWatcher != null && (sDaemon == null || !sDaemon.isAlive())) {
-            initWatcherDaemon();
-        }
-
-        return sWatcher;
-    }
+    private WatchKey mWatchKey;
 
     private static void initWatcherDaemon() {
-        sDaemon = new Thread() {
+        sWatcherDaemon = new Thread() {
             @Override
             public void run() {
-                Log.d(TAG, "Watcher daemon thread started");
+                if (BuildConfig.DEBUG) Log.d(TAG, "Watcher daemon thread started");
                 while (true) {
                     WatchKey key;
                     try {
                         key = sWatcher.take();
-                    } catch (InterruptedException ignored) {
-                        return;
+                    } catch (ClosedWatchServiceException | InterruptedException ignored) {
+                        break;
```

Comment:
I am not sure if continuing execution is the right thing when the thread gets interrupted.
Theoretically, it should only happen when the process is exiting.